src/index.js

```diff
@@ -99,9 +101,37 @@ export function asObject(shape) {
   }
 
   // The shape-aware version:
-  const out = bindObjectShape(shape, false)
+  const keys = Object.keys(shape)
```

Comment:
I see that you have moved the keys to a high-scope for better performance? Idea: lazy-evaluation for the keys; this way we only run-once, but when the cleaner is used (vs when it's defined). What do you think?

Reply:
It's probably not worth it, since this happens once at start-up and never again. Doing it lazily would actually incur a tiny cost every time we clean data, to see if the list exists or not.